exercise/reveal_models.py

```diff
@@ -94,3 +82,19 @@ def get_reveal_time(self, state: 'BaseRevealState') -> Optional[datetime.datetim
                 return latest_deadline + datetime.timedelta(minutes=self.delay_minutes or 0)
 
         return None
+
+    def _is_revealed_due_to_full_points(self, state: 'BaseRevealState') -> bool:
+        points = state.get_points()
+        max_points = state.get_max_points()
+        return points is not None and max_points is not None and points >= max_points
+
+    def _is_revealed_due_to_time(self, state: 'BaseRevealState', time: datetime.datetime) -> bool:
```

Comment:
The type hint for `time` is wrong. Since it can be `None`, the type is `Optional[datetime.datetime]`